packages/block-editor/src/components/block-list/index.js

```diff
@@ -22,6 +22,8 @@ import useBlockDropZone from '../use-block-drop-zone';
  * to avoid laginess.
  */
 const BLOCK_ANIMATION_THRESHOLD = 200;
+// List of blocks to 'spotlight' when active in Full Site Editing.
+const ENTITY_AREAS = [ 'core/template-part' ];
```

Comment:
I do like how this feels more extensible without adding much extra programming complexity

Reply:
I don't like this personally, it's static and references blocks that might or might not exist in the available blocks. Can't we rely on the "controlled" inner blocks instead, we have a flag for that I believe.

Reply:
> might or might not exist in the available blocks

I guess that is one reason running the code that uses this is gated behind the FSE experiment being active for now.

> Can't we rely on the "controlled" inner blocks instead, we have a flag for that I believe.

If we only want to test this for template parts currently, and have the ability to test/add it with other entity controller blocks one at a time, evaluating based on controlled inner blocks would be a bit too general?

Reply:
If we are moving the mapSelect logic to a selector, Im thinking this list of allowed blocks to enable this feature will go there with it.

Reply:
> evaluating based on controlled inner blocks would be a bit too general?

Right now, the only "controlled" blocks are "Post content" or "Template part" and I believe later we'll have "Reusable blocks". All these blocks seem like good candidates for this behavior for me.

Reply:
Checking if it's an inner block controller does make sense to me. I personally haven't been sure how far we should rely on that for this information, but I do like the simplicity.

The only challenge would be if a 3rd party user creates a block which is a controller, but doesn't expect this UI side effect. I don't think that needs to be a big concern for now, though.

Reply:
Regarding detecting by controlled inner blocks - while I do think that would clean things up and that the other controlled inner blocks are good candidates for this behavior, I don't think the design discussions have gotten much of a chance to think about this.

Moreover, detecting by a list of allowed blocks allows us to easily re-use the new selector in other areas that have different needs.  For example, I cleaned up the `packages/edit-site/src/components/header/document-actions/index.js` `useSecondaryText` hook quite a bit by being able to reuse this logic.  In that component, it was argued to only show the title for active template parts and not post content.

So if we wanted to try this spotlight behavior for post content we could just update the selection to be
`getActiveBlockIdByBlockNames( [ 'core/template-part', 'core/post-content' ] )`, yet still have the previously mentioned area just using `getActiveBlockIdByBlockNames( [ 'core/template-part' ] )` (or vice versa).

This format also allows for future usage outside controlled blocks in general.  The new selector could be used for other blocks such as group block, columns block, or whatever else might be useful in future development.

Reply:
I still disagree about having this list in `block-editor` personally. If this is to be a curated list, I think it needs to be a setting of `BlockEditorProvidder` but to be honest, I don't see why design would not want post content to not use spotlight mode like template part, in some ways, it's just a "content" template part. 

If you prefer we can start with an unstable block editor setting?

Reply:
> If you prefer we can start with an unstable block editor setting?

I think that would definitely be better than what we are currently doing.  Would that be to define the list [in the store defaults](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/store/defaults.js) with an `__experimental` prefix?

> I don't see why design would not want post content to not use spotlight mode like template part, in some ways, it's just a "content" template part.

I agree, maybe we can update this from a list in settings to checking for controlled inner blocks in the future instead if this is desirable.  

@shaunandrews @kellychoffman @jameskoster @dubielzyk @olaolusoga - Has there been much discussion or thought regarding using this sort of spotlight mode for Post Content (and in the future reusable blocks) as well?  Similarly with what shows up as the secondary/selected item in the document-actions name panel in the top bar?  I get the feeling that there should be consistency between what we show in the top-bar and what we highlight with the 'spotlight' mode.

Reply:
@youknowriad - Updated to pass the list into settings from the site editor's initialization, with a default empty array on the block-editors side.  The selector will also return early if there is nothing in that array so we got rid of the check for the experiment being active as well. (thanks to some help from @noahtallen 🎉 )